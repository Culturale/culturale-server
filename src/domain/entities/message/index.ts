export * from "./message";
export * from "./message.interface";
export * from "./message.schema";
