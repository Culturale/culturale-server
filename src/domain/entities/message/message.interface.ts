import type { MongoId } from '~/types/types';

export interface IMessage {
  _id: MongoId;
  content: string;
  userId: string;
  date: Date;
}
