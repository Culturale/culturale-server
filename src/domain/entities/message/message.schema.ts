import { Schema, model } from "mongoose";

import type { IMessage } from "./message.interface";

const MessageSchema = new Schema({
  content: { required: false, type: String },
  data: { required: true, type: Date },
  userId: { required: true, type: String },
});

const MessageModel = model<IMessage>("Message", MessageSchema);

export { MessageModel };
