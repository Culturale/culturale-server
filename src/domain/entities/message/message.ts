import type { MongoId } from '~/types/types';

import type { IMessage } from './message.interface';

export class Message implements IMessage {
  public _id: MongoId;
  public content: string;
  public userId: string;
  public date: Date;

  constructor(id: MongoId, content: string, userId: string, date: Date) {
    this._id = id;
    this.content = content;
    this.userId = userId;
    this.date = date;
  }
}
