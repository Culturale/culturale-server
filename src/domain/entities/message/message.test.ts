import { Types } from 'mongoose';

import { Message } from './message';
import type { IMessage } from './message.interface';

describe('Chat Entity', function () {
  let instance: IMessage;

  beforeEach(function () {
    instance = new Message(
      new Types.ObjectId('6423f909d626b03e800a9858'),
      'test-content',
      'test-userId',
      new Date(2),
    );
  });

  it('can be created', function () {
    expect(instance).toMatchSnapshot();
  });
});
