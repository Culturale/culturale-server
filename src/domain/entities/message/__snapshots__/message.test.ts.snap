// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Entity can be created 1`] = `
Message {
  "_id": "6423f909d626b03e800a9858",
  "content": "test-content",
  "date": 1970-01-01T00:00:00.002Z,
  "userId": "test-userId",
}
`;
